<launch>

	<arg name="quad" default="SQ01s"/>
	<arg name="goal_topic" default="goal"/>


	<node ns="$(arg quad)" name="faster" pkg="faster" type="faster_node" output="screen" clear_params="true" > 
    <rosparam file="$(find faster)/param/faster.yaml" command="load"/>
    <!-- <rosparam file="$(find global_mapper_ros)/cfg/global_mapper.yaml" command="load" ns="mapper"/> -->

        <!-- Subscriptions -->
		<remap from="~state" to="state"/>
		<remap from="~term_goal" to="term_goal" />

		<!-- Publications -->
		<remap from="~traj" to="traj"/>
		<remap from="~goal" to="$(arg goal_topic)" />
		<remap from="~setpoint" to="setpoint" />
    <remap from="~log_topic" to="log_topic" />



	</node>

	<node ns="$(arg quad)" name="predictor" pkg="faster" type="predictor_node" output="screen" clear_params="true" > 
		<rosparam file="$(find faster)/param/faster.yaml" command="load" ns="faster"/>

		<remap from="~state" to="state"/>


	</node>


	 <node ns="$(arg quad)" pkg="jsk_rviz_plugins"
        type="rosconsole_overlay_text.py"
        respawn="false"
        name="rosconsole_overlay_text">

        <param name="nodes" value="/$(arg quad)/faster" /> <!--Names of the nodes whose ros output is going to pipe to RViz -->
        <param name="reverse_lines" value="false" /> <!--Names of the nodes whose ros output is going to pipe to RViz -->

<!--     <rosparam>
      reverse_lines: false
      nodes: "$(arg quad)/faster"
    </rosparam> -->
  </node>

</launch>


<!-- launch-prefix="xterm -e gdb -ex run args" -->